openmoc/plotter.py

```diff
@@ -54,12 +59,17 @@
 # @endcode
 #
 # @param track_generator the TrackGenerator which has generated Tracks
-def plot_tracks(track_generator):
-
-    global subdirectory
+# @param get_figure whether or not to return the Matplotlib figure
+def plot_tracks(track_generator, get_figure=False):
 
+    global subdirectory, font
     directory = openmoc.get_output_directory() + subdirectory
 
+    # Ensure that normal settings are used even if called from ipython
+    curr_rc = dict(matplotlib.rcParams)
+    matplotlib.rcParams.update(matplotlib.rcParamsDefault)
+    plt.rc('font', **font)
```

Comment:
You mean store a deepcopy of the rcParams from matplotlib as a global
variable? What would this accomplish? All I can see it doing is saving us
2-3 limes of code since we could inject our font styles into it in one
place. Perhaps that's worth it, just wondered what value you see in doing
this.
On Feb 4, 2016 11:55 PM, "Geoffrey Gunow" <notifications@github.com> wrote:

> In openmoc/plotter.py
> <https://github.com/mit-crpg/OpenMOC/pull/260#discussion_r51977634>:
>
> >      directory = openmoc.get_output_directory() + subdirectory
> >
> > +    # Ensure that normal settings are used even if called from ipython
> > +    curr_rc = dict(matplotlib.rcParams)
> > +    matplotlib.rcParams.update(matplotlib.rcParamsDefault)
> > +    plt.rc('font', **font)
>
> Ok. How about we save the rcParams we want to a class or global variable?
> That way we would just need to set the parameters using that variable.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-crpg/OpenMOC/pull/260/files#r51977634>.
>